contracts/andromeda_ics20/src/contract.rs

```diff
@@ -0,0 +1,335 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, Addr, Binary, Deps, DepsMut, Env, IbcMsg, IbcQuery, MessageInfo, Order,
+    PortIdResponse, Response, StdResult,
+};
+
+use cw2::{get_contract_version, set_contract_version};
+use cw20::{Cw20Coin, Cw20ReceiveMsg};
+
+use crate::amount::Amount;
+use crate::error::ContractError;
+use crate::ibc::Ics20Packet;
+use crate::msg::{
+    ChannelResponse, ExecuteMsg, InitMsg, ListChannelsResponse, MigrateMsg, PortResponse, QueryMsg,
+    TransferMsg,
+};
+use crate::state::{Config, CHANNEL_INFO, CHANNEL_STATE, CONFIG};
+use cw0::{nonpayable, one_coin};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw20-ics20";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InitMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let cfg = Config {
+        default_timeout: msg.default_timeout,
+    };
+    CONFIG.save(deps.storage, &cfg)?;
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Receive(msg) => execute_receive(deps, env, info, msg),
+        ExecuteMsg::Transfer(msg) => {
+            let coin = one_coin(&info)?;
+            execute_transfer(deps, env, msg, Amount::Native(coin), info.sender)
+        }
+    }
+}
+
+pub fn execute_receive(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    wrapper: Cw20ReceiveMsg,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+
+    let msg: TransferMsg = from_binary(&wrapper.msg)?;
+    let amount = Amount::Cw20(Cw20Coin {
+        address: info.sender.to_string(),
+        amount: wrapper.amount,
+    });
+    let api = deps.api;
+    execute_transfer(deps, env, msg, amount, api.addr_validate(&wrapper.sender)?)
+}
+
+pub fn execute_transfer(
+    deps: DepsMut,
+    env: Env,
+    msg: TransferMsg,
+    amount: Amount,
+    sender: Addr,
+) -> Result<Response, ContractError> {
+    if amount.is_empty() {
+        return Err(ContractError::NoFunds {});
+    }
+    // ensure the requested channel is registered
+    // FIXME: add a .has method to map to make this faster
```

Comment:
Is there anything stopping us from making this change instead of leaving it like this?

Reply:
Reference: https://docs.rs/cw-storage-plus/latest/src/cw_storage_plus/map.rs.html#78-80